python/lsst/meas/algorithms/matcherSourceSelector.py

```diff
@@ -162,4 +162,68 @@ def _isUsable(self, source):
             and not source.get(self.fluxFlagKey) \
             and self._goodSN(source)
 
+
+class MatcherPessimisticSourceSelectorTask(MatcherSourceSelectorTask):
+    """
+    !Select sources that are useful for matching.
+
+    Good matching sources have high signal/noise, are non-blended. They need not
+    be PSF sources, just have reliable centroids. This inherited class adds
+    the removal of saturated, interpolated, and edge_key objects to the set of
+    bad flags. It is a temporary addition designed preserve the source selction
+    used in matchOptimisticB. Once matchPessimisticB is adopted as the default
+    source selector the class will be removed and the saturated, interpoalted, and
+    edge_key flags will be added to the matcherSourceSelector class.
+
+    TODO: Once DM-10399 is complete an RFC will be filed to make matchPessimisticB
+    the default matcher this class will replace matcherSourceSelector with this source
+    selector resulting in only one matcherSourceSeletor.
```

Comment:
Created this new ticket DM-10800 in anticipation of the future adoption of matchPessimisticB.